oximeter/db/tests/integration_test.rs

```diff
@@ -0,0 +1,380 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use anyhow::Context;
+use clickward::{Deployment, KeeperClient, KeeperError};
+use dropshot::test_util::log_prefix_for_test;
+use omicron_test_utils::dev::poll;
+use omicron_test_utils::dev::test_setup_log;
+use oximeter::test_util;
+use oximeter_db::{Client, DbWrite, OxqlResult, Sample, TestDbWrite};
+use std::default::Default;
+use std::time::Duration;
+
+pub struct TestInput {
+    n_projects: usize,
+    n_instances: usize,
+    n_cpus: usize,
+    n_samples: usize,
+}
+
+impl Default for TestInput {
+    fn default() -> Self {
+        TestInput { n_projects: 2, n_instances: 2, n_cpus: 2, n_samples: 50 }
+    }
+}
+
+impl TestInput {
+    fn n_timeseries(&self) -> usize {
+        self.n_projects * self.n_instances * self.n_cpus
+    }
+
+    fn n_points(&self) -> usize {
+        self.n_projects * self.n_instances * self.n_cpus * self.n_samples
+    }
+}
+
+#[tokio::test]
+async fn test_cluster() -> anyhow::Result<()> {
+    usdt::register_probes().unwrap();
+    let request_timeout = Duration::from_secs(15);
+    let start = tokio::time::Instant::now();
+    let logctx = test_setup_log("test_cluster");
+    let log = &logctx.log;
+
+    let (parent_dir, prefix) = log_prefix_for_test(logctx.test_name());
+    let path = parent_dir.join(format!("{prefix}-oximeter-clickward-test"));
+    std::fs::create_dir(&path)?;
+
+    let mut deployment = Deployment::new_with_default_port_config(
+        path.clone(),
+        "oximeter_cluster".to_string(),
+    );
+
+    let num_keepers = 3;
+    let num_replicas = 2;
+    deployment
+        .generate_config(num_keepers, num_replicas)
+        .context("failed to generate config")?;
+    deployment.deploy().context("failed to deploy")?;
+
+    let client1 = Client::new_with_request_timeout(
+        deployment.http_addr(1)?,
+        log,
+        request_timeout,
+    );
+    let client2 = Client::new_with_request_timeout(
+        deployment.http_addr(2)?,
+        log,
+        request_timeout,
+    );
+    wait_for_ping(&client1).await?;
+    wait_for_ping(&client2).await?;
+    wait_for_keepers(&deployment, (1..=3).collect()).await?;
+    let end = tokio::time::Instant::now();
+    println!("deploy setup time = {:?}", end - start);
```

Comment:
FWIW, you can use `start.elapsed()` here, it's kinda nice.

Reply:
Oh, nice! I totally forgot about that :)

Reply:
Done in 82f3072109ff6433dad05a371bde8137331b7bff